### PR TITLE
[SandboxVec][DAG] Boilerplate

### PR DESCRIPTION
**Pull request by vporpo**
This patch adds a very basic implementation of the Dependency Graph to be used by the vectorizer.

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  for (auto &I : drop_begin(*BB)) {
+    auto *N = getOrCreateNode(&I);
+    N->addMemPred(LastN);
+    LastN = N;
+  }
+}
+
+#ifndef NDEBUG
+void DependencyGraph::print(raw_ostream &OS) const {
+  // InstrToNodeMap is unoredred so we need to create an ordered vector.
```
Comment (slackito) [inline llvm/lib/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.cpp:47]: typo: unordered

**Comment by vporpo** [comment]
I added a brief description at the top of the header file.

**Review by slackito — APPROVED** [review]

File: llvm/lib/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+#ifndef NDEBUG
+void DependencyGraph::print(raw_ostream &OS) const {
+  // InstrToNodeMap is unordered so we need to create an ordered vector.
+  SmallVector<DGNode *> Nodes;
+  Nodes.reserve(InstrToNodeMap.size());
+  for (const auto &Pair : InstrToNodeMap)
+    Nodes.push_back(Pair.second.get());
+  // Sort them based on which one comes first in the BB.
+  stable_sort(Nodes, [](DGNode *N1, DGNode *N2) {
```
Comment (aeubanks) [inline llvm/lib/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.cpp:53]: doesn't need to be stable if all elements are unique

Comment (vporpo) [reply llvm/lib/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.cpp:53]: true

**Pull request merged**